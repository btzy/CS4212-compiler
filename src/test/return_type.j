class Main {

Void main(Int i, Int a, Int b,Int d){
    i = a;
}
}
class Dummy {
    Int x;
    Void e() { x = x; return;} // ok
    Int f() { x = x; return;} // error
    Int g() { x = x; return "123";} // error
    Void h() { x = x; return "123";} // error
}