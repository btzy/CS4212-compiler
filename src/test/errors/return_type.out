Error: Return statement with no value is not allowed in method that returns "Int" ( line 13 col 16 - line 13 col 22 ) :
13 |         x = x; return;
   |                ^^^^^^^
Note: Method declared to return "Int" here ( line 12 col 5 - line 12 col 7 ) :
12 |     Int f() {
   |     ^^^
Error: Returning a value of type "String" is not allowed in method that returns "Int" ( line 16 col 16 - line 16 col 28 ) :
16 |         x = x; return "abc";
   |                ^^^^^^^^^^^^^
Note: Method declared to return "Int" here ( line 15 col 5 - line 15 col 7 ) :
15 |     Int g() {
   |     ^^^
Error: Returning a value of type "String" is not allowed in method that returns "Void" ( line 19 col 16 - line 19 col 28 ) :
19 |         x = x; return "def";
   |                ^^^^^^^^^^^^^
Note: Method declared to return "Void" here ( line 18 col 5 - line 18 col 8 ) :
18 |     Void h() {
   |     ^^^^
Semantic checking aborted due to above errors.
